radixdlt-core/radixdlt/src/test/java/com/radixdlt/statecomputer/radixengine/RadixEngineShutdownTest.java

```diff
@@ -0,0 +1,198 @@
+import com.radixdlt.statecomputer.forks.ForksModule;
+import com.radixdlt.statecomputer.forks.RERulesConfig;
+import com.radixdlt.statecomputer.forks.RadixEngineForksLatestOnlyModule;
+import com.radixdlt.statecomputer.forks.modules.MainnetForksModule;
+import com.radixdlt.store.DatabaseLocation;
+import com.radixdlt.store.LastStoredProof;
+import com.radixdlt.utils.PrivateKeys;
+import com.radixdlt.utils.UInt256;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public final class RadixEngineShutdownTest {
+  private static final ECKeyPair VALIDATOR_KEY = PrivateKeys.ofNumeric(1);
+
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  @Inject private RadixEngine<LedgerAndBFTProof> radixEngine;
+
+  // FIXME: Hack, need this in order to cause provider for genesis to be stored
+  @Inject @LastStoredProof private LedgerProof ledgerProof;
+
+  private Injector createInjector() {
+    return Guice.createInjector(
+        MempoolConfig.asModule(1000, 10),
+        new MainnetForksModule(),
+        new RadixEngineForksLatestOnlyModule(
+            RERulesConfig.testingDefault()
+                .overrideFeeTable(
+                    FeeTable.create(
+                        Amount.zero(), Map.of(TokenResource.class, Amount.ofTokens(1))))),
+        new ForksModule(),
+        new SingleNodeAndPeersDeterministicNetworkModule(VALIDATOR_KEY, 0),
+        new MockedGenesisModule(
+            Set.of(VALIDATOR_KEY.getPublicKey()),
+            Amount.ofTokens(100_000_000_000L),
+            Amount.ofTokens(100L)),
+        new AbstractModule() {
+          @Override
+          protected void configure() {
+            bindConstant()
+                .annotatedWith(DatabaseLocation.class)
+                .to(folder.getRoot().getAbsolutePath());
+          }
+        });
+  }
+
+  @Test
+  public void no_transactions_can_be_executed_after_the_shutdown_fork() throws Exception {
+    createInjector().injectMembers(this);
+
+    // Creating a token before engine shutdown should work
+    final var txnConstructedAndExecutedBeforeShutdown =
+        radixEngine
+            .construct(createMutableTokenTxnReq(VALIDATOR_KEY, UInt256.TEN, "t1"))
+            .signAndBuild(VALIDATOR_KEY::sign);
+
+    final var res = radixEngine.execute(List.of(txnConstructedAndExecutedBeforeShutdown));
+    assertEquals(1L, res.getProcessedTxns().size());
+
+    // Construct before shutdown but execute after
+    final var txnConstructedButNotExecutedBeforeShutdown =
+        radixEngine
+            .construct(createMutableTokenTxnReq(VALIDATOR_KEY, UInt256.TEN, "t2"))
+            .signAndBuild(VALIDATOR_KEY::sign);
+
+    radixEngine.shutdown();
+
+    // Can't execute an already constructed txn
+    final var res2 = radixEngine.execute(List.of(txnConstructedButNotExecutedBeforeShutdown));
```

Comment:
Why do we return an empty list when executing but throws an exception when constructing?

Reply:
I've changed that to an exception